test/test_core.py

```diff
@@ -7506,21 +7504,21 @@ def test_embind_val_cross_thread_deleted(self):
     self.do_runf('test_embind_val_cross_thread.cpp')
 
   def test_embind_val_coro(self):
-    create_file('post.js', r'''Module.onRuntimeInitialized = () => {
+    create_file('pre.js', r'''Module.onRuntimeInitialized = () => {
```

Comment:
Why the conversion from post.js to pre.js here and below?

Reply:
Adding onRuntimeInitialized during post.js doesn't work in the case we do top-level-await on the wasm module because in that case (and also in the case of sync instantiation) the initialization has already been done before the post-js stuff runs.

Reply:
This changes fixes this test so that it works in all mode.